### DISCUSSION TITLE
Issue Running Local OP Mainnet Development Environment

### DISCUSSION THREAD

**Discussion by doukhahmed** [ethereum-optimism/developers#585] [Application Development]
### Did you check the documentation?

- [X] Yes

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

Hi everyone,

I'm trying to follow the tutorial for setting up a local development environment for the entire OP Mainnet system as described here. However, I'm encountering issues when running the make devnet-up command. I have installed all the necessary dependencies, including Go, but the build fails with several errors related to missing packages in the GOROOT directory (package cmp is not in GOROOT, package slog is not in GOROOT, etc.).

Has anyone encountered a similar issue or knows how to resolve this? Any guidance would be greatly appreciated!

Thanks in advance!
![Sans titre](https://github.com/user-attachments/assets/5a01a17f-02f4-43f0-b76f-7c9dd3b65f27)

**Comment by imtipi** [comment]
`make devnet-up` works for me
can you try `go env` and check the value of `GO111MODULE` ?

**Reply by doukhahmed** [reply]
it's empty 
![Uploading Sans titre.png…]()

**Reply by imtipi** [reply]
can you reupload the screenshot?can't see it

**Reply by doukhahmed** [reply]
@imtipi 
![capture](https://github.com/user-attachments/assets/56c4b3b2-fa7a-4544-8245-8d3a89031a95)

**Reply by imtipi** [reply]
> @imtipi ![capture](https://private-user-images.githubusercontent.com/47477494/365268917-56c4b3b2-fa7a-4544-8245-8d3a89031a95.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU2NDczNjQsIm5iZiI6MTcyNTY0NzA2NCwicGF0aCI6Ii80NzQ3NzQ5NC8zNjUyNjg5MTctNTZjNGIzYjItZmE3YS00NTQ0LTgyNDUtOGQzYTg5MDMxYTk1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTA2VDE4MjQyNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRjYzc1ZjQ1NTYzZjIwNmY1MmYyMGZmNGQyOTRjZWFjZTRjMDIyYmI4ODQ0YTY4YTA1ZDM2MDYzZmM5OTA5ZjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.8xYPfKYokhxpbp3pzDEuzAtnrdRoEH6FdtP3gSKm5BY)

can you run `make mod-tidy` then try make devnet -up again?

**Reply by imtipi** [reply]
> ![Capture d’écran 2024-09-06 223043](https://private-user-images.githubusercontent.com/47477494/365310437-64f03618-08c7-496f-b93e-7ae82fa49a8a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU3MTYzMjUsIm5iZiI6MTcyNTcxNjAyNSwicGF0aCI6Ii80NzQ3NzQ5NC8zNjUzMTA0MzctNjRmMDM2MTgtMDhjNy00OTZmLWI5M2UtN2FlODJmYTQ5YThhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTA3VDEzMzM0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFmODcyODk0MmEzNDVjOTRhNzJhNmI3YjVmODNiOWU4MmJkYWE5ZjU0NzBmYTkxNTQ0MGNhZTNlNDY1OTNmYTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.UtPevNJVEG4F96sRVxfKvneVLFKRmgnBMFPvSaXYnKo)

wait,can you do `go version`?seems it is too old
edit:i saw you are 1.18,better do a update and try make devnet-up again

**Reply by doukhahmed** [reply]
i try go version : 
go version go1.18.1 linux/amd64

**Reply by imtipi** [reply]
> i try go version : go version go1.18.1 linux/amd64

please update your go

**Comment by doukhahmed** [comment]


**Reply by imtipi** [reply]
do `make mod-tidy` again

**Reply by doukhahmed** [reply]
![Sans titre](https://github.com/user-attachments/assets/ff83e014-5dfc-4844-984c-d06a3be5245c)

**Reply by doukhahmed** [reply]
![Sans titre](https://github.com/user-attachments/assets/1a3e9f99-ee04-4e16-987d-ed41960aa553)

**Reply by imtipi** [reply]
please follow the guide
https://go.dev/doc/install
seems you didn't delete the old version of go so error still occur

**Reply by doukhahmed** [reply]
![Sans titre](https://github.com/user-attachments/assets/f79a15d1-071f-4b23-b213-7e2b82a36419)
i remove the old version an i keep 1.23.1 version ...

**Reply by imtipi** [reply]
remember the remove command in go.dev just removed all version of go, so reextract again

**Reply by doukhahmed** [reply]
![Sans titre](https://github.com/user-attachments/assets/4dca6242-d73e-4ff1-93cf-334ec79f4e6b)

**Reply by imtipi** [reply]
do `sudo export PATH=$PATH:/usr/local/go/bin` ,seems root environment doesn't know the exist of go

**Reply by doukhahmed** [reply]
![Sans titre](https://github.com/user-attachments/assets/a4c6af80-2dea-4be2-afd4-21c72f716c9e)

**Reply by doukhahmed** [reply]
when i try sudo export PATH=$PATH:/usr/local/go/bin i got  "sudo: export: command not found"

**Reply by imtipi** [reply]
ah,can you do `sudo which go` ,if go actually not executable,then try add `export PATH=$PATH:/usr/local/go/bin` into `/etc/profile` then `source /etc/profile` and try again

**Reply by doukhahmed** [reply]
but export PATH=$PATH:/usr/local/go/bin it pass

**Reply by imtipi** [reply]
> when i try sudo export PATH=$PATH:/usr/local/go/bin i got "sudo: export: command not found"

my wrong,should edit /etc/profile directly,cannot do sudo export

**Reply by imtipi** [reply]
> but export PATH=$PATH:/usr/local/go/bin it pass

it only apply to current user,not root user(also sudo)

**Reply by doukhahmed** [reply]
can we make a call ?

**Reply by doukhahmed** [reply]
https://meet.google.com/asp-htrz-eys

**Reply by imtipi** [reply]
> can we make a call ?

sorry,i personal can't,can you ask in discord?i know someone from our folks can do it
also cc @opfocus @joohhnnn

**Reply by doukhahmed** [reply]
not prob. 
"ah,can you do sudo which go ,if go actually not executable,then try add export PATH=$PATH:/usr/local/go/bin into /etc/profile then source /etc/profile and try again" 
just add export PATH=$PATH:/usr/local/go/bin  in the file /etc/profile ?

**Reply by imtipi** [reply]
> not prob. "ah,can you do sudo which go ,if go actually not executable,then try add export PATH=$PATH:/usr/local/go/bin into /etc/profile then source /etc/profile and try again" just add export PATH=$PATH:/usr/local/go/bin in the file /etc/profile ?

yes,maybe also need `sudo source /etc/profile` after edit

**Reply by imtipi** [reply]
no,you must edit `/etc/profile`,~ means in your current user,will not apply to sudo

**Comment by opfocus** [comment]
Please check your go version, go 1.21 is recommended

**Reply by imtipi** [reply]
i said you need to add export to /etc/profile,you can try `sudo go version` to see if go is exist in sudo/root environment

**Reply by opfocus** [reply]
Are you using a virtual machine?

**Reply by doukhahmed** [reply]
yes with ubuntu 22.04

**Reply by doukhahmed** [reply]
@opfocus

**Reply by opfocus** [reply]
Would you consider not using a virtual machine?

**Comment by nic225** [comment]
<img width="676" alt="image" src="https://github.com/user-attachments/assets/8396dd7c-01f2-4614-b178-c7c1621e4bd3">

Have you guys seen this error when running make devnet-up, I have also tried using the latest go version and no luck

**Reply by nic225** [reply]
@imtipi @opfocus

**Reply by nic225** [reply]
Thanks guys!

**Reply by opfocus** [reply]
I think it is caused by the L1 beacon not starting successfully.
What you may need to check is whether the go/bin path is added to the environment variable and takes effect.

**Reply by nic225** [reply]
Hi thanks opfocus, 

I added go to the $GOROOT and $GOPATH but that did not seem to do the trick.

<img width="683" alt="image" src="https://github.com/user-attachments/assets/61a3046f-caae-495c-b094-636f40ccb133">

**Reply by opfocus** [reply]
`export PATH=$PATH:/usr/local/go/bin`   I mean like this    
Add an environment variable like this and make it take effect

**Reply by iethena** [reply]
i have the same issue

**Reply by nic225** [reply]
I got it working thank you opfocus

**Reply by doukhahmed** [reply]
@opfocus i got this error 
`$ sudo make devnet-up
[sudo] password for adoukh: 
git submodule update --init --recursive
make[1]: Entering directory &apos;/home/adoukh/optimism&apos;
./ops/scripts/geth-version-checker.sh &amp;&amp; \
 	(echo &quot;Geth versions match, not installing geth...&quot;; true) || \
 		(echo &quot;Versions do not match, installing geth!&quot;; \
 			go install -v [github.com/ethereum/go-ethereum/cmd/geth@v1.14.7](http://github.com/ethereum/go-ethereum/cmd/geth@v1.14.7); \
 			echo &quot;Installed geth!&quot;; true)
./ops/scripts/geth-version-checker.sh: line 7: geth: command not found
Geth version does not match!
Local geth version: v
Expected geth version: v1.14.7-stable
Versions do not match, installing geth!
Installed geth!
make[1]: Leaving directory &apos;/home/adoukh/optimism&apos;
make[1]: Entering directory &apos;/home/adoukh/optimism&apos;
go install -v [github.com/protolambda/eth2-testnet-genesis@v0.10.0](http://github.com/protolambda/eth2-testnet-genesis@v0.10.0)
make[1]: Leaving directory &apos;/home/adoukh/optimism&apos;
./ops/scripts/newer-file.sh .devnet/allocs-l1.json ./packages/contracts-bedrock \
	|| make devnet-allocs
make[1]: Entering directory &apos;/home/adoukh/optimism&apos;
git submodule update --init --recursive
make[2]: Entering directory &apos;/home/adoukh/optimism&apos;
./ops/scripts/geth-version-checker.sh &amp;&amp; \
 	(echo &quot;Geth versions match, not installing geth...&quot;; true) || \
 		(echo &quot;Versions do not match, installing geth!&quot;; \
 			go install -v [github.com/ethereum/go-ethereum/cmd/geth@v1.14.7](http://github.com/ethereum/go-ethereum/cmd/geth@v1.14.7); \
 			echo &quot;Installed geth!&quot;; true)
./ops/scripts/geth-version-checker.sh: line 7: geth: command not found
Geth version does not match!
Local geth version: v
Expected geth version: v1.14.7-stable
Versions do not match, installing geth!
Installed geth!
make[2]: Leaving directory &apos;/home/adoukh/optimism&apos;
make[2]: Entering directory &apos;/home/adoukh/optimism&apos;
go install -v [github.com/protolambda/eth2-testnet-genesis@v0.10.0](http://github.com/protolambda/eth2-testnet-genesis@v0.10.0)
make[2]: Leaving directory &apos;/home/adoukh/optimism&apos;
PYTHONPATH=./bedrock-devnet python3 ./bedrock-devnet/main.py --monorepo-dir=. --allocs
Traceback (most recent call last):
  File &quot;/home/adoukh/optimism/./bedrock-devnet/main.py&quot;, line 9, in &lt;module&gt;
    main()
  File &quot;/home/adoukh/optimism/./bedrock-devnet/main.py&quot;, line 5, in main
    devnet.main()
  File &quot;/home/adoukh/optimism/bedrock-devnet/devnet/__init__.py&quot;, line 97, in main
    devnet_l1_allocs(paths)
  File &quot;/home/adoukh/optimism/bedrock-devnet/devnet/__init__.py&quot;, line 137, in devnet_l1_allocs
    run_command([
  File &quot;/home/adoukh/optimism/bedrock-devnet/devnet/__init__.py&quot;, line 366, in run_command
    return subprocess.run(
  File &quot;/usr/lib/python3.10/subprocess.py&quot;, line 503, in run
    with Popen(*popenargs, **kwargs) as process:
  File &quot;/usr/lib/python3.10/subprocess.py&quot;, line 971, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File &quot;/usr/lib/python3.10/subprocess.py&quot;, line 1863, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: &apos;forge&apos;
make[1]: *** [Makefile:204: devnet-allocs] Error 1
make[1]: Leaving directory &apos;/home/adoukh/optimism&apos;
make: *** [Makefile:182: devnet-up] Error 2`

**Reply by opfocus** [reply]
This seems to be a similar problem to the one above, go installs the programs and you can see them in the go/bin directory. But because this directory is not added to the environment variables, the script cannot use them.
Add the directory where they exist to the environment variable and make it effective

**Reply by doukhahmed** [reply]
@opfocus thanks, i added the directory to the environment variables ..but now i get this error i think the version of geth alright ? 
`adoukh@adoukh-VirtualBox:~/optimism$ sudo make devnet-up
git submodule update --init --recursive
make[1]: Entering directory '/home/adoukh/optimism'
./ops/scripts/geth-version-checker.sh && \
 	(echo "Geth versions match, not installing geth..."; true) || \
 		(echo "Versions do not match, installing geth!"; \
 			go install -v [github.com/ethereum/go-ethereum/cmd/geth@v1.14.7](http://github.com/ethereum/go-ethereum/cmd/geth@v1.14.7); \
 			echo "Installed geth!"; true)
./ops/scripts/geth-version-checker.sh: line 7: geth: command not found
Geth version does not match!
Local geth version: v
Expected geth version: v1.14.7-stable
Versions do not match, installing geth!
Installed geth!
make[1]: Leaving directory '/home/adoukh/optimism'
make[1]: Entering directory '/home/adoukh/optimism'
go install -v [github.com/protolambda/eth2-testnet-genesis@v0.10.0](http://github.com/protolambda/eth2-testnet-genesis@v0.10.0)
make[1]: Leaving directory '/home/adoukh/optimism'
./ops/scripts/newer-file.sh .devnet/allocs-l1.json ./packages/contracts-bedrock \
	|| make devnet-allocs
make[1]: Entering directory '/home/adoukh/optimism'
git submodule update --init --recursive
make[2]: Entering directory '/home/adoukh/optimism'
./ops/scripts/geth-version-checker.sh && \
 	(echo "Geth versions match, not installing geth..."; true) || \
 		(echo "Versions do not match, installing geth!"; \
 			go install -v [github.com/ethereum/go-ethereum/cmd/geth@v1.14.7](http://github.com/ethereum/go-ethereum/cmd/geth@v1.14.7); \
 			echo "Installed geth!"; true)
./ops/scripts/geth-version-checker.sh: line 7: geth: command not found
Geth version does not match!
Local geth version: v
Expected geth version: v1.14.7-stable
Versions do not match, installing geth!
Installed geth!
make[2]: Leaving directory '/home/adoukh/optimism'
make[2]: Entering directory '/home/adoukh/optimism'
go install -v [github.com/protolambda/eth2-testnet-genesis@v0.10.0](http://github.com/protolambda/eth2-testnet-genesis@v0.10.0)
make[2]: Leaving directory '/home/adoukh/optimism'
PYTHONPATH=./bedrock-devnet python3 ./bedrock-devnet/main.py --monorepo-dir=. --allocs
Traceback (most recent call last):
  File "/home/adoukh/optimism/./bedrock-devnet/main.py", line 9, in <module>
    main()
  File "/home/adoukh/optimism/./bedrock-devnet/main.py", line 5, in main
    devnet.main()
  File "/home/adoukh/optimism/bedrock-devnet/devnet/__init__.py", line 97, in main
    devnet_l1_allocs(paths)
  File "/home/adoukh/optimism/bedrock-devnet/devnet/__init__.py", line 137, in devnet_l1_allocs
    run_command([
  File "/home/adoukh/optimism/bedrock-devnet/devnet/__init__.py", line 366, in run_command
    return subprocess.run(
  File "/usr/lib/python3.10/subprocess.py", line 503, in run
    with Popen(*popenargs, **kwargs) as process:
  File "/usr/lib/python3.10/subprocess.py", line 971, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/usr/lib/python3.10/subprocess.py", line 1863, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'forge'
make[1]: *** [Makefile:204: devnet-allocs] Error 1
make[1]: Leaving directory '/home/adoukh/optimism'
make: *** [Makefile:182: devnet-up] Error 2`

**Reply by opfocus** [reply]
`[Errno 2] No such file or directory: 'forge' make[1]`
https://docs.optimism.io/chain/testing/dev-node#install-foundry-using-monorepo-tooling

**Reply by doukhahmed** [reply]
when i try forge --version i got the right version ... 
![Sans titre](https://github.com/user-attachments/assets/a426d7aa-d74d-44c1-8051-aef3e0ec425b)

**Reply by opfocus** [reply]
Sorry, if you are using a virtual machine, I can't provide any valuable information.
Yes, I'm skeptical about virtual machines being able to run devent.

**Comment by nic225** [comment]
https://github.com/ethereum-optimism/optimism/issues/11428#issuecomment-2397227915

Have you guys seen this issue referenced here. I am trying to run optimism on a server separate from my m2 mac now. @opfocus @imtipi

**Reply by opfocus** [reply]
https://github.com/ethereum-optimism/developers/discussions/17#discussioncomment-10728373  I ran it with a similar configuration here and had no issues.  If you continue to have this problem, please let me know

**Reply by nic225** [reply]
Oh yes sorry for the other comment, I am still experiencing this issue. Other notes. I am running docker in rootless mode, I have `export PATH="$PATH:$HOME/go/bin"` ubuntu 24.04.1 with go version go1.23.2 linux/amd64 could lowering the go version make this work, I don't have admin access so that would be delayed.

**Reply by nic225** [reply]
<img width="1320" alt="image" src="https://github.com/user-attachments/assets/480c04d8-1226-43ba-9398-844d0da6f71d">
Further up, I keep getting these addresses initialzied as 0x000000..., which is probably not correct.

**Reply by nic225** [reply]
There is an open issue here, sorry just noticed it now. https://github.com/ethereum-optimism/optimism/issues/12354

**Reply by opfocus** [reply]
thank you for this info
